### ISSUE TITLE
Capture Console?

### ISSUE THREAD

**Issue by johnaweiss** [ChromeDevTools/awesome-chrome-devtools#109]
Beginner question:

Is it possible to capture all chrome console messages using javascript?
This would be a stand-alone chrome browser, _not_ headless.
Running inside the same browser-instance-- not a client/remote browser. 

What's the simplest, most lightweight solution?

thx!

**Comment by Garbee** [comment]
I don't believe there is any exposure to intercept those messages. What is your use-case for needing such a feature?

**Comment by johnaweiss** [comment]
Display them on a front-end IDE, embedded in a webpage, using the local JS engine. 

On 6/2/2020 7:05:59 AM, Jonathan Garbee <notifications@github.com> wrote:
I don't believe there is any exposure to intercept those messages. What is your use-case for needing such a feature?

**Comment by Garbee** [comment]
IDEs tend to connect using remote debugging which allows API access via the [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/). That way your IDE *is* the debugger for the app rather than having DevTools open and your IDE. That's how they tend to get access to the logs.

Would using the remote debugging setup not work for some reason?

**Comment by Yash-Singh1** [comment]
I had a similar problem. My solution was to edit the console functions: https://github.com/Yash-Singh1/zoom.js/blob/master/test/auto.js#L3

**Comment by o0101** [comment]
Hey @johnaweiss and @Yash-Singh1 this will probably be too late for you but for anyone else who wants this in future: you could proxy the debugger front-end. 
I have created a project that does that, [here](https://github.com/dosyago/devtoolium/blob/548d4b5e8111edb49fdf0bc71d7174d8165bdd88/src/index.js#L302): [devtoolium](https://github.com/dosyago/devtoolium)

You could fork this and implement your own interception
